src/mkinitcpio.rs

```diff
@@ -0,0 +1,139 @@
+use crate::parse_args::AppConfig;
+use crate::utils::csleep;
+use rand::prelude::*;
+use rand::seq::SliceRandom;
+use yansi::Paint;
+
+const ALL_HOOKS: &'static [&str] = &[
+    &"base",
+    &"udev",
+    &"usr",
+    &"resume",
+    &"autodetect",
+    &"modconf",
+    &"block",
+    &"net",
+    &"mdadm",
+    &"mdadm_udev",
+    &"keyboard",
+    &"keymap",
+    &"consolefont",
+    &"encrypt",
+    &"lvm2",
+    &"fsck",
+    &"filesystems",
+];
+
+const REQUIRED_HOOKS: &'static [&str] = &[
+    &"base",
+    &"udev",
+    &"autodetect",
+    &"modconf",
+    &"block",
+    &"fsck",
+    &"filesystems",
+];
+
+const ZIP_FORMATS: &'static [&str] = &[&"gzip", &"bzip2", &"lzma", &"xz", &"lzop", &"lz4"];
+
+fn warn(msg: &str) {
+    println!(
+        "{}{}",
+        Paint::yellow("==> WARNING: ").bold(),
+        Paint::new(msg).bold()
+    );
+}
+
+fn msg1(msg: &str) {
+    println!("{}{}", Paint::green("==> ").bold(), Paint::new(msg).bold());
+}
+
+fn msg2(msg: &str) {
+    println!("{}{}", Paint::blue("  -> ").bold(), Paint::new(msg).bold());
+}
+
+fn build(hooks: &[&str], preset: &str, mode: &str, zip: &str, appconfig: &AppConfig) {
+    let mut rng = thread_rng();
+
+    msg1(
+        format!(
+            "Building image from preset: /etc/mkinitcpio.d/{preset}.preset: '{mode}'",
+            preset = preset,
+            mode = mode
+        )
+        .as_ref(),
+    );
+
+    let image = format!(
+        "/boot/initramfs-{preset}{suffix}.img",
+        preset = preset,
+        suffix = if mode == "default" {
+            "".to_string()
+        } else {
+            format!("-{}", mode)
+        }
+    );
+
+    msg2(
+        format!(
+            "-k /boot/vmlinuz-{preset} -c /etc/mkinitcpio.conf -g {image}",
+            preset = preset,
+            image = image
+        )
+        .as_ref(),
+    );
+    msg1("Starting build: 5.0.7-arch1-1-ARCH");
```

Comment:
Would be cool to vary this a bit.

Reply:
Found some [here](https://en.wikipedia.org/wiki/Uname). Put them in a file and randomized. :tada: